File: testtableview/src/main/java/me/box/app/testtableview/impl/IContext.java
/*
 * Copyright © 2017 CHANGLEI. All rights reserved.
 */

package me.box.app.testtableview.impl;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by box on 2017/5/23.
 *
 * 自己实现的上下文环境
 */
public interface IContext {

    void onInitViews(Bundle savedInstanceState);

    void onInitDatas(Bundle savedInstanceState);

    void onInitListeners(Bundle savedInstanceState);

    View getLayout(Bundle savedInstanceState, LayoutInflater inflater, ViewGroup parent);
}

File: testtableview/src/main/java/me/box/app/testtableview/impl/IContext.kt
package me.box.app.testtableview.impl

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by box on 2017/5/23.
 *
 * 自己实现的上下文环境
 */
interface IContext {

    fun onInitViews(savedInstanceState: Bundle?)

    fun onInitDatas(savedInstanceState: Bundle?)

    fun onInitListeners(savedInstanceState: Bundle?)

    fun getLayout(savedInstanceState: Bundle?, inflater: LayoutInflater, parent: ViewGroup): View?
}